docs/docker-optimization/comparison.md

```
@@ -0,0 +1,263 @@
+# Before & After Comparison
+
+## Stage Structure Changes
+
+### BEFORE (5 stages)
```

markdownlint-cli2 (0.18.1)

Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)